docs/tutorials/extending_admin_ui/3_creating_a_content_list.md

```diff
@@ -0,0 +1,209 @@
+                'label' => 'Content List',
+                'route' => 'ezsystems_extending_tutorial.all_content_list.list',
+            ]
+        );
+    }
+}
+```
+
+This listener subscribes to the `ConfigureMenuEvent::MAIN_MENU` event (see line 14).
+
+Line 28 points to the new route that you need to add to the routing file.
+
+## Add routing
+
+Add the following block to `src/EzSystems/ExtendingTutorialBundle/Resources/config/routing.yml`:
+
+``` yaml hl_lines="5"
+ezsystems_extending_tutorial.all_content_list.list:
+    path: /all_content_list/{page}
+    defaults:
+        page: 1
+        _controller: 'EzSystemsExtendingTutorialBundle:AllContentList:list'
+```
+
+## Create a controller
+
+As you can see in the code above, the next step is creating a controller that will take care of the article list view.
+
+Firs, ensure that the controller is configured in `services.yml`.
+Add the following block to that file:
+
+``` yaml
+EzSystems\ExtendingTutorialBundle\Controller\:
+    resource: "../../Controller/*"
+    autowire: true
+    autoconfigure: true
+    public: false
+    exclude: "../../Controller/{Controller}"
+```
+
+Then, in `src/EzSystems/ExtendingTutorialBundle/Controller` create a `AllContentListController.php` file:
+
+```php hl_lines="41"
+<?php
+
+namespace EzSystems\ExtendingTutorialBundle\Controller;
+
+use EzSystems\EzPlatformAdminUiBundle\Controller\Controller;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\Core\Pagination\Pagerfanta\LocationSearchAdapter;
+use Pagerfanta\Pagerfanta;
+
+class AllContentListController extends Controller
+{
+    private $searchService;
+
+    private $contentTypeService;
+
+    public function __construct(SearchService $searchService, ContentTypeService $contentTypeService)
+    {
+        $this->searchService = $searchService;
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    public function listAction($page = 1)
+    {
+        $query = new LocationQuery();
+
+        $criterions = [
+            new Criterion\Visibility(Criterion\Visibility::VISIBLE),
+        ];
+
+        $query->query = new Criterion\LogicalAnd($criterions);
+
+        $paginator = new Pagerfanta(
+            new LocationSearchAdapter($query, $this->searchService)
+        );
+        $paginator->setMaxPerPage(8);
+        $paginator->setCurrentPage($page);
+
+        return $this->render('@EzSystemsExtendingTutorial/list/all_content_list.html.twig', [
+            'totalCount' => $paginator->getNbResults(),
+            'articles' => $paginator,
+        ]);
+    }
+}
+```
+
+The highlighted line 41 indicates the template that will be used to display the list.
+
+## Add a template
+
+Finally, create an `all_content_list.html.twig` file in `src/EzSystems/ExtendingTutorialBundle/Resources/views/list`:
```

Comment:
I suggest  "create the file" instead of "create a file", because it has a name.

Reply:
But it's not a "the" file before we've created it...